### PR TITLE
Ensure generated CI mocks are up to date

### PR DESCRIPTION
**Pull request by Lemma199**
**Description**

Adds make targets to generate mocks, and adds configuration to run checks in CI to make sure generated mocks are included in a pull request.

**Tests**

No tests for make and CI stuff 🤷 

**Invariants**

N/A

**Additional context**

N/A

**Metadata**

- Fixes #[[54](https://github.com/ethereum-optimism/ecosystem-contributions/issues/54)]

**TODOs**

- [x] Author or reviewer has added an entry to the [current release notes draft][RND], if appropriate. (**N/A**)

[RND]: https://www.notion.so/oplabs/ded30107ceec41c88817e60322aa8d0a?v=b4a22cedb85a46a38c9be14e7c984953&pvs=4 "Release Notes"

### CODE REVIEW

File: .circleci/config.yml
Comment (tynes) [inline .circleci/config.yml:1547]: looks like the spacing is off here

Comment (Lemma199) [reply .circleci/config.yml:1547]: I think the diff is just cut off making it look that way, see below. I.e. line `1546` is part of the `requires` section of another job in the `main` workflow.
https://github.com/ethereum-optimism/optimism/blob/be65577495e6e19f111beaa9eb3295c585821989/.circleci/config.yml#L1538-L1548

I added the new jobs to the end of the list, but if it's preferable to have them grouped with other relevant jobs I'm happy to move them around!

File: .circleci/config.yml
Comment (trianglesphere) [inline .circleci/config.yml:1151]: CI builder doesn't have `mockery` installed

Comment (Lemma199) [reply .circleci/config.yml:1151]: Yep: https://github.com/ethereum-optimism/optimism/pull/5839#issuecomment-1570398060

I can install it in the `check-generated-mocks-op-node` job before running generate, or alternatively add it to the `ci-builder` container. I'll see if I (outside collaborator) can update the docker container and if CI publishes the images.

Comment (Lemma199) [reply .circleci/config.yml:1151]: I opened #5844 to add mockery to the `ci-builder` container. Once that PR is good to go and merged, we can re-run CI in this PR.

**Comment by smartcontracts** [comment]
Ok just waiting on a build/publish of the ci-builder image! Will keep you updated.

**Comment by smartcontracts** [comment]
@Lemma199 can you rebase this on the latest develop? Mockery should now be included in the image.

**Comment by Lemma199** [comment]
Thanks @smartcontracts, done!

**Comment by Lemma199** [comment]
Failing due to some diffs (basically comments):
```diff
-// Code generated by mockery v2.23.1. DO NOT EDIT.
+// Code generated by mockery v2.28.1. DO NOT EDIT.
```

There's also some code being removed (presumably stale):
```diff
diff --git a/op-node/p2p/mocks/GossipMetricer.go b/op-node/p2p/mocks/GossipMetricer.go
index 38ebf8d50..d5da64382 100644
--- a/op-node/p2p/mocks/GossipMetricer.go
+++ b/op-node/p2p/mocks/GossipMetricer.go
@@ -1,4 +1,4 @@
-// Code generated by mockery v2.22.1. DO NOT EDIT.
+// Code generated by mockery v2.28.1. DO NOT EDIT.
 
 package mocks
 
@@ -14,11 +14,6 @@ func (_m *GossipMetricer) RecordGossipEvent(evType int32) {
 	_m.Called(evType)
 }
 
-// SetPeerScores provides a mock function with given fields: _a0
-func (_m *GossipMetricer) SetPeerScores(_a0 map[string]float64) {
-	_m.Called(_a0)
-}
-
```

I can apply these changes and push up a new commit that includes them — I couldn't find what version (if any) documentation for this repository suggests to use for `mockery`, but moving forward CI would block a PR generating mocks if the user is not using the same version as the `ci-builder` container. Is that alright? Alternatively I can modify the CI task to run with `—disable-version-string` which wouldn't update the these version strings.

**Comment by trianglesphere** [comment]
>  couldn't find what version (if any) documentation for this repository suggests to use for mockery

We don't have one. It was pretty recently added & hasn't been standardized. Individual devs have been responsible for keeping everything up to date. It's probably fine to requires a specific version if we let people know.

**Comment by ajsutton** [comment]
The `GossipMetricer.go` updates will cause some knock-on problems in `peer_scorer_test.go` which should actually be using a `ScoreMetrics` mock rather than `GossipMetricer` - I suspect some refactoring happened and because the mock wasn't updated, the test continued working. I ran into it with https://github.com/ethereum-optimism/optimism/pull/5870/ so it has the updates to `peer_scorer_test` required to make it work. I think it's probably better to pull the required changes in here since that's what they're really related to. I have some more testing to do on that other PR before it's ready to merge anyway and it won't be hard for me to sort out conflicts.

Good sign we need this check - thanks for picking it up. :)

**Comment by Lemma199** [comment]
Thanks for the heads up @ajsutton. I'm happy to wait for #5870 to be merged and then can rebase this PR, I don't mind leaving this open.

**Comment by ajsutton** [comment]
That merge conflict is because #5870 just merged which should resolve the `GossipMetricer` mock issues.  There may be other mocks out of date as well but hopefully just regenerating them with the right version will be all that's required to get it all green.

**Comment by Lemma199** [comment]
Fixed the conflicts and updated the mocks using `2.28.1`. Should be good to go now! Thanks @ajsutton

**Review by ajsutton — APPROVED** [review]
LGTM. Thanks for this.

**Review by trianglesphere — APPROVED** [review]

**Pull request merged**
